com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ShaderUtil.java

```diff
@@ -246,12 +247,27 @@ public static Result transform(String input, ShaderType shaderType, String targe
                 input = input.replaceAll("\\b" + keywordRep[0] + "\\b", keywordRep[1]);
             }
 
+            // Get data output array size
+            Matcher fragDataArrayMatcher = regexFragDataArrayPattern.matcher(input);
+            int maxColorOutputs = 1;
+            while (fragDataArrayMatcher.find()) {
+                String fragDataArrayIndex = fragDataArrayMatcher.group("index");
+                maxColorOutputs = Math.max(maxColorOutputs, Integer.parseInt(fragDataArrayIndex) + 1);
+            }
+
+            // Replace fragment output variables 
             boolean output_glFragColor = input.contains(glFragColorKeyword);
             boolean output_glFragData = input.contains(glFragDataKeyword);
+
             if (output_glFragColor)
-                input = input.replaceAll("\\b" + glFragColorKeyword + "\\b", glFragColorRep);
+            {
+                input = input.replaceAll("\\b" + glFragColorKeyword + "\\b", glFragColorRep + "_0");
+            }
+
             if (output_glFragData)
-                input = input.replaceAll("\\b" + glFragDataKeyword + "\\b", glFragColorRep);
+            {
+                input = input.replaceAll("\\b" + glFragDataKeyword + "\\[(\\d+)\\]", glFragColorRep + "_$1");
+            }
 
```

Comment:
Previously the gl_FragData keyword would be replaced with the same color output as gl_FragColor, but that doesn't work for multiple outputs. Instead we generate specific outputs for each assignment of gl_FragData[x] into gl_FragData_x variables.